### PR TITLE
Remove sqlite dependency

### LINKED ISSUE #22146: RFC: deprecate old graphdb migration / move to an external tool?
Just some thought; the mattn/go-sqlite3 package is a big vendored package, and only used [to migrate the legacy graphdb](https://github.com/docker/docker/blob/19a453e6b40177af31703f107131aae5ca81adf2/daemon/daemon.go#L321-L341) links (Docker 1.9 and older) to the new "runtime" links (https://github.com/docker/docker/pull/16032)

Should we;
- mark the option to migrate from 1.9 versions as "deprecated" (to be removed in 1.14)?
- move the migration to an external tool?

### PR DESCRIPTION
**Pull request by justincormack**
fixes #22146
relates to https://github.com/moby/moby/pull/16032

Rebase of #25782 cc @LK4D4 

- What I did
Removed sqlite dependency.
- How I did it
Removed every usage of graphdb.
- How to verify it
Compile daemon
- Description for the changelog

Dependency from sqlite removed, so links upgrade from 1.9 does not work. External migration tool should be used if necessary.

cutest-cat-picture-ever

### CODE REVIEW

**Comment by cpuguy83** [comment]
SGTM.
Maybe we should add a note to the changelog about not being able to upgrade directly from <= 1.9... or maybe just something like `*note*: It is recommended to upgrade only one version at a time`

**Comment by LK4D4** [comment]
Maybe we should add a warning to logs if old links DB exists. It doesn't require SQLite and small enough I think.

**Comment by cpuguy83** [comment]
Well, we kept the db around in case ppl wanted to rollback.

File: contrib/builder/deb/amd64/generate.sh [hunk 1]
```diff
@@ -76,7 +76,6 @@ for version in "${versions[@]}"; do
 		libdevmapper-dev # for "libdevmapper.h"
 		libltdl-dev # for pkcs11 "ltdl.h"
 		libseccomp-dev  # for "seccomp.h" & "libseccomp.so"
-		libsqlite3-dev # for "sqlite3.h"
```
Comment (tophj-ibm) [inline contrib/builder/deb/amd64/generate.sh]: don't forget to remove it from all the other builders we have now too.

Comment (justincormack) [inline contrib/builder/deb/amd64/generate.sh]: ah, I thought I had but you are right it didn't actually generate them as I was on amd64...

Comment (justincormack) [inline contrib/builder/deb/amd64/generate.sh]: added these.

**Comment by justincormack** [comment]
re-pushed to unstick CI

**Comment by LK4D4** [comment]
Ok, design looks good to me
ping @docker/core-maintainers for move to code-review

**Review by LK4D4 — APPROVED** [review]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**